### ISSUE TITLE
Allow pinning setup-trivy by hash

### ISSUE THREAD

**Issue by smola** [aquasecurity/trivy-action#423]
I work with repos where all GitHub Actions are pinned by commit hash, and passlisted in the repo settings. As of https://github.com/aquasecurity/trivy-action/pull/406, it is not possible to do this anymore, since it internally fetches `aquasecurity/setup-trivy` by tag, and there does not seem to be any way to override that. Even if `skip-setup-trivy: true` is set, it seems GitHub will require passlisting the tag version.

**Comment by simar7** [comment]
> Even if skip-setup-trivy: true is set, it seems GitHub will require passlisting the tag version.

Could you expand more on this, where does this passlisting happen? Is it a GitHub requirement?

**Comment by smola** [comment]
This could also be fixed by using a commit hash reference in action.yml for setup-trivy. Users who do not restrict their referenced Actions will not notice the difference. Those who enforce commit hash pinning will be able to do so.

**Comment by assignUser** [comment]
Yes pinning any used actions to SHA would be appreciated, as mentioned this action is otherwise not usable in orgs that enforce sha pinning through the allowlist.

Pinning used actions to hash is also a security best practice [recommended by GitHub](https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions#using-third-party-actions). Given recent compromises of actions through force-pushed tags this is not only best practice but seems to be the only safe way to use external actions.
